src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlInternalConnectionTds.cs

```diff
@@ -1911,15 +1912,27 @@ private void ResolveExtendedServerName(ServerInfo serverInfo, bool aliasLookup,
         {
             SqlClientEventSource.Log.TryAdvancedTraceEvent("<sc.SqlInternalConnectionTds.AttemptOneLogin|ADV> {0}, timout={1}[msec], server={2}", ObjectID, timeout.MillisecondsRemaining, serverInfo.ExtendedServerName);
             RoutingInfo = null; // forget routing information 
+            string localDBDataSource = null;
 
+            if (ConnectionOptions.LocalDBInstance != null)
+            {
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    localDBDataSource = LocalDB.GetLocalDBDataSource(serverInfo.UserServerName,timeout);
+                }
```

Comment:
This checking bypasses the unsupported platforms and any inner implementation to check them will remain untouched.

Reply:
Added Unsupported exception and unit test for it.